platform/darwin/src/MGLStyleValue_Private.h

```diff
@@ -214,7 +214,7 @@ class MGLStyleValueTransformer {
 
     // Color
     void getMBGLValue(MGLColor *rawValue, mbgl::Color &mbglValue) {
-        mbglValue = rawValue.mgl_color;
+        mbglValue = rawValue.mgl_colorForPremultipliedValue;
```

Comment:
I think this function may be dead code that’s just here because mbgl’s template system requires it to satisfy the type constraints. When I evaluate a color expression at runtime on either iOS or macOS, we seem to go down a different code path:

```
#0	0x0000000100494366 in ::-[NSColor(MGLAdditions) mgl_color]() at /path/to/mapbox-gl-native-ios/platform/macos/src/NSColor+MGLAdditions.mm:21
#1	0x0000000100451125 in ::-[NSColor(MGLExpressionAdditions) mgl_jsonExpressionObject]() at /path/to/mapbox-gl-native-ios/platform/darwin/src/NSExpression+MGLAdditions.mm:483
#2	0x00000001004651f5 in ::-[NSExpression(MGLAdditions) mgl_jsonInterpolationExpressionObject]() at /path/to/mapbox-gl-native-ios/platform/darwin/src/NSExpression+MGLAdditions.mm:1332
#3	0x0000000100460e3d in ::-[NSExpression(MGLAdditions) mgl_jsonExpressionObject]() at /path/to/mapbox-gl-native-ios/platform/darwin/src/NSExpression+MGLAdditions.mm:1146
#4	0x00000001003c79f7 in std::__1::enable_if<!(std::is_same<mbgl::style::PropertyValue<mbgl::Color>, mbgl::style::ColorRampPropertyValue>::value), mbgl::style::PropertyValue<mbgl::Color> >::type MGLStyleValueTransformer<mbgl::Color, NSColor* __strong, mbgl::Color, NSColor* __strong>::toPropertyValue<mbgl::style::PropertyValue<mbgl::Color> >(NSExpression*, bool) at /path/to/mapbox-gl-native-ios/platform/darwin/src/MGLStyleValue_Private.h:94
#5	0x0000000100499e6f in ::-[MGLFillStyleLayer setFillColor:](NSExpression *) at /path/to/mapbox-gl-native-ios/platform/darwin/src/MGLFillStyleLayer.mm:115
```

This code path seems to corroborate https://github.com/mapbox/mapbox-gl-native/pull/16153#issuecomment-580416201, which suggests that returning premultiplied color components in `MGLColor.mgl_jsonExpressionObject` would address the issue.

Alternatively, we could modify `UIColor.mgl_color`/`NSColor.mgl_color` instead, but that method is also used in a couple other places, namely the annotation API. Given https://github.com/mapbox/mapbox-gl-native/issues/7524#issuecomment-461881639, it’s unclear to me whether the annotation API – which is based on the runtime styling API – still expects premultiplied color components, as this very old comment suggests:

https://github.com/mapbox/mapbox-gl-native/blob/f105100943478eee209c7044c3054fc158bef0a6/platform/ios/src/UIColor%2BMGLAdditions.mm#L9-L11